randovania/game_description/resources/inventory.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import TYPE_CHECKING, NamedTuple, Self
+
+from randovania.game_description.resources.resource_type import ResourceType
+
+if TYPE_CHECKING:
+    from randovania.game_description.resources.item_resource_info import ItemResourceInfo
+    from randovania.game_description.resources.resource_info import ResourceCollection, ResourceGain
+
+
+class InventoryItem(NamedTuple):
+    amount: int
+    capacity: int
+
+
+@dataclasses.dataclass(frozen=True, slots=True)
+class Inventory:
+    raw: dict[ItemResourceInfo, InventoryItem]
```

Comment:
should this have a default factory to default to an empty dict? or is the explicit `empty()` initializer preferable?

Reply:
I think I prefer calling `.empty`. This will need a better `__init__.py` at some point, but I'll leave that refactor to the future